spec/forms/ctc/w2s/misc_info_form_spec.rb

```diff
@@ -52,6 +61,41 @@
     expect(w2.box13_statutory_employee).to eq 'yes'
     expect(w2.box13_retirement_plan).to eq 'yes'
     expect(w2.box13_third_party_sick_pay).to eq 'no'
+    expect(w2.w2_box14.other_description).to eq "description"
+    expect(w2.w2_box14.other_amount).to eq 12
+    expect(w2.w2_state_fields_group.box15_state).to eq "NY"
+    expect(w2.w2_state_fields_group.box15_employer_state_id_number).to eq "abcd1234"
+    expect(w2.w2_state_fields_group.box16_state_wages).to eq 1
+    expect(w2.w2_state_fields_group.box17_state_income_tax).to eq 2
+    expect(w2.w2_state_fields_group.box18_local_wages).to eq 3
+    expect(w2.w2_state_fields_group.box19_local_income_tax).to eq 4
+    expect(w2.w2_state_fields_group.box20_locality_name).to eq "someplace"
+  end
+
+  it "updates an existing w2_state_fields_group and w2_box14" do
+    w2.create_w2_state_fields_group(box16_state_wages: "100")
+    w2.create_w2_box14(other_description: 'banana', other_amount: 12)
+    params = {
+      other_description: 'papaya',
+      other_amount: '24',
+      box15_state: "NY",
+      box15_employer_state_id_number: "abcd1234",
+      box16_state_wages: "1",
+      box17_state_income_tax: "2",
+      box18_local_wages: "3",
+      box19_local_income_tax: "4",
+      box20_locality_name: "someplace",
+    }
+    described_class.new(w2, params).save
+    expect(w2.w2_state_fields_group.box16_state_wages).to eq 1
+    expect(w2.w2_state_fields_group.box15_state).to eq "NY"
+    expect(w2.w2_state_fields_group.box15_employer_state_id_number).to eq "abcd1234"
+    expect(w2.w2_state_fields_group.box17_state_income_tax).to eq 2
+    expect(w2.w2_state_fields_group.box18_local_wages).to eq 3
+    expect(w2.w2_state_fields_group.box19_local_income_tax).to eq 4
+    expect(w2.w2_state_fields_group.box20_locality_name).to eq "someplace"
+    expect(w2.w2_box14.other_description).to eq 'papaya'
+    expect(w2.w2_box14.other_amount).to eq 24
```

Comment:
no duplication could be checked here I guess but since it's non null on w2_id don't see how it could duplicate so that's cool